application/src/test/java/com/sanction/thunder/validation/PropertyValidatorTest.java

```diff
@@ -0,0 +1,198 @@
+package com.sanction.thunder.validation;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class PropertyValidatorTest {
+  private final Map<String, Object> properties = new HashMap<>();
+  private List<PropertyValidationRule> validationRules
+      = Collections.singletonList(
+          new PropertyValidationRule("firstProperty", "string"));
+
+  @Test
+  public void testSkipValidation() {
+    PropertyValidator validator = new PropertyValidator(null);
+
+    assertTrue(validator.isValidPropertiesMap(properties));
+  }
+
+  @Test
+  public void testInvalidSize() {
+    PropertyValidator validator = new PropertyValidator(validationRules);
+
+    assertFalse(validator.isValidPropertiesMap(properties));
+  }
+
+  @Test
+  public void testMismatchName() {
+    PropertyValidator validator = new PropertyValidator(validationRules);
+    properties.put("myProperty", "value");
```

Comment:
Can you make a copy of properties instead of appending so each test is independent. I can see this potentially causing problems later.